### PR TITLE
homework simplified unique_ptr

### PR DESCRIPTION
**Pull request by jakjan95**
Also added missed CI to run UTs and memory check

### CODE REVIEW

File: homework/unique_ptr/include/UniquePtr.hpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            other.ptr_ = nullptr;
+        }
+
+        return *this;
+    }
+
+    ~unique_ptr() noexcept
+    {
+        if (ptr_) {
+            delete ptr_;
```
Comment (WojtekMs) [inline homework/unique_ptr/include/UniquePtr.hpp:42]: This check is redundant, deleting nullptr is safe :)

File: homework/unique_ptr/include/UniquePtr.hpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    explicit operator bool() const noexcept
+    {
+        return ptr_ != nullptr;
+    }
+
+    void reset(T* ptr = nullptr) noexcept
+    {
+        if (ptr_) {
+            delete ptr_;
```
Comment (WojtekMs) [inline homework/unique_ptr/include/UniquePtr.hpp:69]: Same comment as in destructor

File: homework/unique_ptr/include/UniquePtr.hpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        ptr_ = ptr;
+    }
+
+    T* release() noexcept
+    {
+        T* current_pointer = ptr_;
+        ptr_ = nullptr;
+        return current_pointer;
```
Comment (WojtekMs) [inline homework/unique_ptr/include/UniquePtr.hpp:78]: This can be collapsed to single line `return std::exchange(ptr_, nullptr);`

**Review by WojtekMs — COMMENTED** [review]
Good code :)

**Comment by ziobron** [comment]
Thanks @WojtekMs for Code Review.
🏅 1 XP granted

**Pull request closed without merging**